src/controls/ControlPanel.js

```diff
@@ -645,8 +649,13 @@ class ControlPanel extends React.Component {
   handleChange(control) {
     let updateName = false
     let { isCustomName } = this.props
-    const { controlData, originalControlData } = this.props
+    const { controlData, originalControlData, setSelectedTemplate } = this.props
     const { id: field, type, syncWith, syncedWith } = control
+
+    if(control.id === 'templateName'){
+      setSelectedTemplate(control.active)
+    }
+
```

Comment:
- This looks like it is based on the specific ID of a control rather than a certain type of control.
- The `setSelectedTemplate` property is passed down from `<TemplateEditor>` but seems to apply to one control only. Would it make more sense for the callback to be supplied on the control itself?
- If you do need to be able to supply the callback as a prop to `TemplateEditor`, then have you considered that this mechanism seems to pass the active value from the control to this callback, so it seems like a generically useful feature that perhaps should not be named for this one use-case. Would it make sense to instead define an `onChange` property that receives the full control, or at least `id` and `active`, any time any control is modified?

Reply:
Added an onChange callback prop that takes the control.
https://github.com/open-cluster-management/temptifly/pull/82/files#diff-01d57ad160a39d6ab9af03fd6faeb584f7fac9488c5d7663cb987520065c2a9eR656